Tests/SonarScanner.MSBuild.Common.Test/MsBuildPathSettingsTests.cs

```diff
@@ -34,10 +35,10 @@ public void GetImportBeforePaths_NonWindows_AppData_Is_NullOrEmpty()
         {
             Action action;
 
-            action = new Action(() => new MsBuildPathSettings((folder, o) => string.Empty, IsWindows(false), DirExists(true)).GetImportBeforePaths());
+            action = new Action(() => MsBuildPathSettings(string.Empty, false, _ => true, false).GetImportBeforePaths());
```

Comment:
This is hard to read without an IDE. Consider using named arguments. (Oh well, they are not allowed according to our coding styles, so forget about it).

Reply:
@martin-strecker-sonarsource I haven't found anywhere in the [coding styling rules](https://github.com/SonarSource/sonar-dotnet/blob/master/docs/coding-style.md) a rule preventing their use. Could you point me to the relevant section?

Reply:
It is not in there, but Pavel complained about their usage almost every time I used them.

Reply:
OK, can you then point me to the relevant discussion you had with Pavel?
Alternatively, it would help moving forward if you could state clearly your position as to whether I should use named arguments for the case at hand.

Reply:
No. We had a game of me suggesting using named arguments, and he rejected the idea for quite some time. See, e.g., https://github.com/SonarSource/sonar-dotnet/pull/5965#discussion_r940138163. I can not remember when we started and where it was discussed for the first time. We both have very different opinions on that. I still prefer named arguments for literal arguments, especially `true`, `false`, `0`, `string. Empty`, `null`, and the like.

Reply:
OK, given that we are basically down to preferences, and there are no arguments that go beyond personal taste, I would go with the following approach:
- `true` and `false` for `isWindows` and `isMacOSX` parameters have been replaced with a single enum `OS`. So there is no confusion as to what `true` or `false` binds to when looking at the call site.
-  I have created a `private static readonly Func<string, bool> DirectoryAlwaysExists = _ => true;` and used that, instead of `_ => true`. So there is no confusion as to what is the purpose of the lambda being passed.
- Now that the number of method parameters is three, and two of them are explicit, there is no general need for named arguments.
- I used named arguments as an alternative to `null as string,` as you have suggested because they nicely solve the overloading problem and convey more meaning, making the code more readable.

@martin-strecker-sonarsource  Please resolve this discussion if there is nothing blocking on your side, regarding the approach described above, or express your concerns (beyond personal taste on the matter). I need to have clear answers from the first round, before moving to the second.